tripleblindmarket/covid-safe-paths
# Feature-Rtl : Add Arabic Language + handling Right to left languages UI and App behavior

## Description

As its a global Application we need it to support many languages as we can, and some of these languages are RTL (Right to left) languages.

**This PR contains the following** 
- Add **ar.json** for Arabic language locales 
- Add RTL language files ISO codes to check the selected language is included within the file or not (12 languages)
- Add React-native-restart package to enable app restarting **because** Mobile Apps language direction change require app restart to apply the new language direction to the UI
- Add **DynamicText** as a reusable component, It inherits the react native **Text** but change text direction based on the selected language  **( Should be used for any new text )**
- Add **DynamicTextInput** as a reusable component, It inherits the react native **TextInput** but change text direction based on the selected language  
- Changed All React native **Text** to **DynamicText** and **TextInput** to **DynamicTextInput**
- Add **languageDirectionHandler** function to languages file as it detects and change current language direction
- Add **languageDirectionHandler** to NativePicker to check if any change to language direction to restart the application to apply the new direction
- Add a new Prop to **NativePicker** named **languagePicker** to make the component know when to use the new **languageDirectionHandler** function , **WHY** to reuse the Picker as we want for all types of pickers
- Add RTL or LTR direction styles to each of **Typography** and **NavigationBarWrapper** component

## How to test

1- Now the only RTL language is the Arabic one
2- If you selected RTL language App will ask you to restart if your current direction is LTR, and vice versa
3- All Views should be shifted to the right when you select RTL language
